plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,845 @@
+  # example), its metadata and spec would look like this:
+  #
+  # metadata:
+  #   name: "example-multigres-cluster-us-west-2"
+  #   labels:
+  #     multigres.com/cluster: "example-multigres-cluster"
+  #     multigres.com/topo-scope: "cell"
+  #     multigres.com/cell: "us-west-2"
+  # spec:
+  #   rootPath: "/multigres/us-west-2"
+  #   image: "quay.io/coreos/etcd:v3.5.17"
+  #   replicas: 1
+  #   dataVolumeClaimTemplate:
+  #     resources:
+  #       requests:
+  #         storage: "5Gi"
+status:
+  conditions:
+  - type: Available
+    status: "True"
+    lastTransitionTime: "2025-10-21T10:35:00Z"
+    message: "Etcd cluster is healthy"
+  replicas: 3
+  readyReplicas: 3
+  clientServiceName: "example-multigres-cluster-global-client"
+  peerServiceName: "example-multigres-cluster-global-peer"
+```
+
+
+
+## Cell CR - Read-Only Child of MultigresCluster
+
+
+```yaml
+# This 'Cell' CR is created from an item in the `spec.cells.templates` list
+# in the parent `MultigresCluster`.
+#
+# This specific example is for 'us-east-1', which had no 'topoServer'
+# block, so it defaults to using the 'global' topoServer.
+#
+apiVersion: multigres.com/v1alpha1
+kind: Cell
+metadata:
+  name: "example-multigres-cluster-us-east-1"
+  namespace: multigres
+  labels:
+    multigres.com/cluster: "example-multigres-cluster"
+    multigres.com/cell: "us-east-1"
+  ownerReferences:
+  # The Cell CR is owned by the MultigresCluster
+  - apiVersion: multigres.com/v1alpha1
+    kind: MultigresCluster
+    name: "example-multigres-cluster"
+    uid: "a1b2c3d4-1234-5678-90ab-f0e1d2c3b4a5"
+    controller: true
+spec:
+  # The logical name of the cell, copied from the template.
+  name: "us-east-1"
+
+  # The parent MultigresCluster passes down the relevant
+  # images for this controller to use.
+  images:
+    multigateway: "multigres/multigres:latest"
+    multiorch: "multigres/multigres:latest"
+
+
+  # The multigateway spec is copied from multigrescluster
+  multigateway:
+    replicas: 2
+    resources:
+      requests:
+        cpu: "500m"
+        memory: "512Mi"
+      limits:
+        cpu: "1"
+        memory: "1Gi"
+
+ # The multiorch spec is copied from multigrescluster.
+  multiorch:
+    replicas: 1
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+       limits:
+        cpu: "200m"
+        memory: "256Mi"
+  
+  # A reference to the GLOBAL TopoServer.
+  # This is always populated by the parent controller.
+  globalTopoServer:
+    rootPath: "/multigres/global"
+    clientServiceName: "example-multigres-cluster-global-client"
+
+  # ALTERNATIVE CONFIG: (Using the Global TopoServer)
+  #
+  # Because the 'us-east-1' cell in the parent CR had no 'topoServer'
+  # block, the MultigresCluster controller sets this to 'global'.
+  # The Cell controller will use this to configure its MultiGateway
+  # to talk to the global topo server.
+  topoServer:
+    global:
+      rootPath: "/multigres/global"
```

Comment:
Why do we have `globalTopoServer` and `topoServer` at the top level? Why not a single `topoServer` stanza, and within that, we can have `global` and `local` separately, and local being optional?

Reply:
So both of these fields are inherited from the MultigresCluster so they make sense. The main issue is that there should have been a `global` field under `topoServer` I have changed it [here](https://github.com/numtide/multigres-operator/pull/33/commits/716f51a7562b9f21ec01231d327f6f1745288058)

Now the field is set to empty when no topoServer is configured for the cell.